test/test_helper.rb

```diff
@@ -8,6 +8,23 @@
 
 REQUIRED_METADATA_KEYS = %w[topic short_description].freeze
 
+MAX_TOPIC_LENGTH = 35
+
+TOPIC_REGEX = /\A[a-z0-9][a-z0-9-]*\Z/
+
+def invalid_topic_message(topic)
+  "'#{topic}' must be between 1-#{MAX_TOPIC_LENGTH} characters, start with a letter or number, " \
+    "and may include hyphens"
+end
+
+def valid_topic?(raw_topic)
+  return false unless raw_topic
+  normalized_topic = raw_topic.gsub(/\A[[:space:]]+/, "").gsub(/[[:space:]]+\z/, "")
+                              .gsub(/[[:space:]]+/, " ").downcase.tr(" ", "-").tr("_", "-")
+  normalized_topic.length <= MAX_TOPIC_LENGTH && normalized_topic =~ TOPIC_REGEX &&
+    !normalized_topic.empty?
```

Comment:
Could invert some of these checks to separate them with `return false if` (or unless) which saves your brain combining the logic? Personal style though so feel free to ignore.